### ISSUE TITLE
Cannot choose test configuration to run before commit in a Flutter project using Android Studio

### ISSUE THREAD

**Issue by doctor-g** [flutter/flutter-intellij#6384]
## Steps to Reproduce

To reproduce:

1. Create a new Flutter project using the Flutter plug-in.
2. Initialize a git repository using VCS->Enable Version Control Integration.
3. Run the widget_test.dart test in order to create a test run configuration.
4. Open the commit panel from the bar on the left.
5. Click the gear icon next to Amend to "show commit options".
6. Under "Run Tests", "Choose Configuration" will not show any test configurations, showing "Nothing here" instead.

I've tried adding test configurations to run all the tests in the folder instead, and also adding non-flutter unit tests. Regardless of these steps, I get no options to choose from in the "Choose Configuration" command.

The expected behavior is that one of my run configurations should show up there, such as the one that was created in step 3 above. This is how I have been able to run tests prior to commit in IntelliJ IDEA, so I expected it to work here.

Note that I've reproduced this error on both Windows and Linux. I happen to be on Linux right now.

## Version info

[✓] Flutter (Channel stable, 3.3.6, on Ubuntu 22.04.1 LTS 5.15.0-52-generic, locale en_US.UTF-8)
    • Flutter version 3.3.6 on channel stable at /home/pvg/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (8 days ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /home/pvg/Android/Sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.2)
    • Android Studio at /snap/android-studio/123/android-studio
    • Flutter plugin version 70.0.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.3)
    • Android Studio at /snap/android-studio/124/android-studio
    • Flutter plugin version 71.0.3
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at /home/pvg/apps/idea-IC-203.6682.168
    • Flutter plugin version 68.1.4
    • Dart plugin version 221.5588

[✓] IntelliJ IDEA Community Edition (version 2022.2)
    • IntelliJ at /snap/intellij-idea-community/390
    • Flutter plugin version 69.0.5
    • Dart plugin version 222.3739.24

[✓] VS Code (version 1.72.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.50.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.1 LTS 5.15.0-52-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 107.0.5304.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

**Comment by stevemessick** [comment]
It seems plugin test configs (i.e. `TestConfig`) are not structured the way `RunChecksCheckinHandlerFactory.createConfigurationChooser()` expects. Looks like it needs to inherit from `ConsolePropertiesProvider` or `CompoundRunConfiguration`. Not sure how much work that will be, but will look into it.

**Comment by stevemessick** [comment]
@doctor-g I was able to get this working. However, it probably doesn't do exactly what you expect. The initial commit is pre-processed by IntelliJ to analyze all the files. It uses its own rules for what a legal Android file should look like, not the Flutter rules. As a result, you get a lot of spurious errors reported. Just use `Commit Anyway` to ignore them. I verified that it did run the selected test config before doing the commit. Assuming I don't find any other problems caused by the code, this will be available in the next release. That will be out the first of Feb.

<img width="260" alt="Screenshot 2023-01-06 at 2 38 29 PM" src="https://user-images.githubusercontent.com/8518285/211111842-66b1e0cc-10f9-4b5e-9daf-cdb246e72321.png">
